sonic-xcvrd/xcvrd/xcvrd.py

```diff
@@ -1120,11 +1129,34 @@ def test_datapath_state(self, api, channel, states):
 
         return done
 
+    def get_host_tx_status(self, lport):
+        host_tx_ready = 'False'
+
+        asic_index = self.port_mapping.get_asic_id_for_logical_port(lport)
+        state_port_tbl = self.xcvr_table_helper.get_state_port_tbl(asic_index)
+
+        found, port_info = state_port_tbl.get(lport)
+        if found and 'host_tx_ready' in dict(port_info):
+            host_tx_ready = dict(port_info)['host_tx_ready']
+        return host_tx_ready
+
+    def get_port_admin_status(self, lport):
+        admin_status = 'down'
+
+        asic_index = self.port_mapping.get_asic_id_for_logical_port(lport)
+        cfg_port_tbl = self.xcvr_table_helper.get_cfg_port_tbl(asic_index)
+
+        found, port_info = cfg_port_tbl.get(lport)
+        if found:
+            # Check admin_status too ...just in case
+            admin_status = dict(port_info)['admin_status']
+        return admin_status
+
     def task_worker(self):
-        self.log_notice("Starting...")
+        self.log_notice("Starting CmisTaskMgr with pid:{}".format(self.task_process.pid))
 
         # APPL_DB for CONFIG updates, and STATE_DB for insertion/removal
-        sel, asic_context = port_mapping.subscribe_port_update_event(['APPL_DB', 'STATE_DB'])
+        sel, asic_context = port_mapping.subscribe_port_update_event()
```

Comment:
While APPL_DB is for monitoring config updates arrived at swss#orchagent, the STATE_DB is for the transceiver insertion/removal detection, and it's not necessary to have both listeners registered all the time.
May I have a clarification on why we need this?

Reply:
@ds952811 A new signal "host_tx_ready" is introduce for 400G QSFP-DD transceiver datapath activation as per this [HLD](https://github.com/sonic-net/SONiC/blob/master/doc/sfp-cmis/Interface-Link-bring-up-sequence.md) where the OA will control this signal and Xcvrd will listen to any change of this signal in STATE_DB.